internal/service/alarms/serve.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"log/slog"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	api "github.com/openshift-kni/oran-o2ims/internal/service/alarms/api/generated"
+	"github.com/openshift-kni/oran-o2ims/internal/service/alarms/internal"
+)
+
+func main() {
+	Serve()
+}
+
+// Serve TODO: Call this func using cobra-cli from inside deployment CR.
+func Serve() {
+	// TODO: Init client-go
+
+	// TODO: Init DB client
+
+	// TODO: Audit and Insert data database
+
+	// TODO: Launch k8s job for DB remove archived data
+
+	// Init server
+	// Create the handler
+	alarmServer := internal.AlarmsServer{}
+
+	alarmServerStrictHandler := api.NewStrictHandlerWithOptions(&alarmServer, nil,
+		api.StrictHTTPServerOptions{
+			RequestErrorHandlerFunc:  getOranReqErrFunc(),
+			ResponseErrorHandlerFunc: getOranRespErrFunc(),
+		})
+
+	r := http.NewServeMux()
+
+	opt := api.StdHTTPServerOptions{
+		BaseRouter: r,
+		Middlewares: []api.MiddlewareFunc{ // Add middlewares here
+			internal.AlarmsOapiValidation(),
+			internal.LogDuration(),
+		},
+		ErrorHandlerFunc: getOranReqErrFunc(),
+	}
+
+	// Register the handler
+	api.HandlerWithOptions(alarmServerStrictHandler, opt)
+
+	// Server config
+	srv := &http.Server{
+		Handler:      r,
+		Addr:         net.JoinHostPort("127.0.0.1", "8080"),
+		ReadTimeout:  5 * time.Second,
+		WriteTimeout: 10 * time.Second,
+		IdleTimeout:  120 * time.Second,
+		ErrorLog:     log.New(os.Stdout, "alarms-http-server: ", log.Ldate|log.Ltime|log.Lshortfile),
+	}
+
+	// Channel to listen for errors coming from the listener.
+	serverErrors := make(chan error, 1)
+	// Channel for shutdown signals
+	shutdown := make(chan os.Signal, 1)
+	signal.Notify(shutdown, os.Interrupt, syscall.SIGTERM)
+
+	// Start server
+	go func() {
+		slog.Info(fmt.Sprintf("Listening on %s", srv.Addr))
+		if err := srv.ListenAndServe(); err != nil && !errors.Is(err, http.ErrServerClosed) {
+			serverErrors <- err
+		}
+	}()
+
+	// Blocking select
+	select {
+	case err := <-serverErrors:
+		slog.Error(fmt.Sprintf("error starting server: %s", err))
+	case sig := <-shutdown:
+		slog.Info(fmt.Sprintf("Shutdown signal received: %v", sig))
+		if err := gracefulShutdown(srv); err != nil {
+			slog.Error(fmt.Sprintf("graceful shutdown failed: %v", err))
+		}
+	}
+}
+
+// gracefulShutdown allow graceful shutdown with timeout
+func gracefulShutdown(srv *http.Server) error {
+	// Create shutdown context with 10 second timeout
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Attempt graceful shutdown
+	if err := srv.Shutdown(ctx); err != nil {
+		// TODO: handle this error
+		slog.Error(fmt.Sprintf("failed graceful shutdown: %v", err))
```

Comment:
error is logged outside as well

Reply:
yeah...accidentally removed the return `err`. Added it back and instead wrapped error with additional detail